### PR TITLE
create job tasks

### PR DESCRIPTION
**Pull request by MBifolco**

### CODE REVIEW

File: components/stratum_message/include/stratum_message.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+typedef enum
+{
+    OTHER_UNKNOWN = 20,
+    JOB_NOT_FOUND = 21,
+    DUPLICATE_SHARE = 22,
+    LOW_DIFFICULTY_SHARE = 23,
+    UNAUTHRIZED_WORKER = 24,
+    NOT_SUBSCRIBED = 25,
+    NUM_ERRORS
+} response_error_code;
```
Comment (juannyG) [inline components/stratum_message/include/stratum_message.h:33]: response from server or back to server?

File: components/stratum_message/include/stratum_message.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    char *result; // can by anything I guess
+    cJSON *error;
+} stratum_server_response_message;
+
+typedef struct
+{
+    char *error_code; //number or string
+    char *message;
+    char *data; // optional and specified as 'object'
+} parased_stratum_server_response_message_error;
```
Comment (juannyG) [inline components/stratum_message/include/stratum_message.h:54]:
```suggestion
} parsed_stratum_server_response_message_error;
```

File: components/stratum_message/include/stratum_message.h [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+typedef struct
+{
+    char *job_id; //The job ID for the job being sent in this message.
+    char *prev_block_hash; //The hex-encoded previous block hash.
+    char *coinbase_prefix; //The hex-encoded prefix of the coinbase transaction (to precede extra nonce 2).
+    char *coinbase_suffix; //The hex-encoded suffix of the coinbase transaction (to follow extra nonce 2).
+    uint8_t *merkle_branches; //A JSON array containing the hex-encoded hashes needed to compute the merkle root. See Merkle Tree Hash Array.
```
Comment (juannyG) [inline components/stratum_message/include/stratum_message.h:62]: will also need "number of merkle_branches" :heart:

File: components/utils/utils.c [hunk 4]
```diff
@@ -0,0 +1,311 @@
+#include "utils.h"
```
Comment (juannyG) [inline components/utils/utils.c:1]: Include a comment in this file citing where we got these from :heart:

File: components/stratum_message/include/stratum_message.h (hunk 3 above)
Comment (juannyG) [reply components/stratum_message/include/stratum_message.h:62]: oh - interesting...yeah i'm fine with either/or

**Comment by MBifolco** [comment]
Let me know if you want to walk through

**Pull request merged**